plugins/autolink/plugin.js

```diff
@@ -9,12 +9,9 @@
 	var doubleQuoteRegex = /"/g;
 
 	CKEDITOR.plugins.add( 'autolink', {
-		requires: 'clipboard,textmatch',
+		requires: 'clipboard,textmatch,link',
 
 		init: function( editor ) {
-			var urlTemplate = new CKEDITOR.template( '<a href="{link}">{text}</a>' ),
```

Comment:
@Comandeer templates have been introduced because in the previous version `autolink` doesn't require link plugin which is a bug thus `autolink` doesn't make sense without `link`. With the `link` plugin, it's easy to create a new link using its API, so we can safely drop manual DOM creation by templates. I understand your point that templating **may** be useful in the future, but currently, we don't have a ticket for such a feature and I think it's not worth to keep bloated code "just in case".

How having two-step link creation improves readability if we can explicitly set link text and attribute instead of reading HTML template?

Reply:
After second thought it makes sense… but it would be nice to have user-modifiable templates. Maybe some day… ;)